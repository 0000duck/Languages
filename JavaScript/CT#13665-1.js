/*
    https://notepad-plus-plus.org/community/topic/13665/function-list-doesn-t-show-classes-in-js-ecmascript
*/

class Scene {

    constructor() {
        this.status = "inactive";
    }

    activate() {
        if (this.status === "inactive") {
            this.status = "active";
        }
    }

    deactivate() {
        if (this.status === "active") {
            this.status = "inactive";
        }
    }
}
