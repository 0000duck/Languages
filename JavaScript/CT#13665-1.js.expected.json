{
	"root":"CT#13665-1.js",
	"branches":[
		{
			"name":"Scene",
			"leaves":[
				"constructor",
				"activate",
				"deactivate"
			]
		}
	]
}